internal/providers/docker/compose/yaml.go

```diff
@@ -160,12 +162,114 @@ type Service struct {
 	// TODO: ulimits
 	User string `yaml:"user"`
 	// TODO: userns_mode
-	Volumes []string `yaml:"volumes"` // TODO: support long syntax.
+	Volumes []VolumeMount `yaml:"volumes"`
 	// TODO: volumes_from
 
 	WorkingDir string `yaml:"working_dir"`
 }
 
+type VolumeMount struct {
```

Comment:
for sub structures with substantial parsing logic, i've been breaking those out in to their own files. this file probably should be called `project.go` instead of `yaml.go`, but I just never got around to renaming it as the contents outgrew the file :)